bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/servlets/DownloadServlet.java

```diff
@@ -95,16 +110,59 @@ private boolean checkForInlineSelector(SlingHttpServletRequest request) {
         return bytes;
     }
 
+    @SuppressWarnings("UnstableApiUsage")
     private void sendResponse(byte[] bytes, Asset asset, SlingHttpServletResponse response, boolean inline) throws IOException {
         response.setContentType(asset.getMimeType());
         response.setContentLength(bytes.length);
         if (inline) {
-            response.setHeader(CONTENT_DISPOSITION, "inline");
+            response.setHeader(CONTENT_DISPOSITION_HEADER, "inline");
         } else {
-            response.setHeader(CONTENT_DISPOSITION, "attachment; filename=\"" + asset.getName() + "\"");
+            response.setHeader(CONTENT_DISPOSITION_HEADER, "attachment; filename=\"" + asset.getName() + "\"");
         }
+        response.setHeader("ETag", ByteSource.wrap(bytes).hash(Hashing.md5()).toString().toUpperCase());
```

Comment:
In this case I would only send last-modified as calculating the MD5 on a large blob might consume considerable time/memory. And in general one mechanism should be enough.